### PR TITLE
feat/generic block improvements

### PR DESCRIPTION
**Pull request by gcornut**
- feat(generic-block): improve default prop values
- feat(generic-block): remove sections if empty
- feat(generic-block): prevent content overflow by default

### CODE REVIEW

File: packages/lumx-react/src/components/generic-block/GenericBlock.stories.tsx [hunk 1]
```diff
@@ -49,6 +49,22 @@ export const Vertical = ({ theme }: HasTheme) => (
...
+export const VerticalOverflow = ({ theme }: HasTheme) => (
+    <div style={{ width: 300 }}>
+        <GenericBlock orientation={Orientation.vertical}>
+            <GenericBlock.Figure style={redBorderStyle}>
+                <Icon icon={mdiPencil} size={Size.m} />
+            </GenericBlock.Figure>
+            <GenericBlock.Content style={redBorderStyle}>
+                _________________________________________________________________
+            </GenericBlock.Content>
+            <GenericBlock.Actions fillSpace style={redBorderStyle}>
```
Comment (Mcourt) [inline packages/lumx-react/src/components/generic-block/GenericBlock.stories.tsx:61]: Do we want the actions to `fillSpace` by default? It looks like they always are in the stories.

**Review by Mcourt — APPROVED** [review]
Small comment but otherwise LGTM

File: packages/lumx-react/src/components/generic-block/GenericBlock.stories.tsx (hunk 1 above)
Comment (gcornut) [reply packages/lumx-react/src/components/generic-block/GenericBlock.stories.tsx:61]: No actually here `fillSpace` does not have any effect in vertical mode without height constraint ([see storybook](https://5fbfb1d508c0520021560f10-tcmubuhvry.chromatic.com/?path=/story/lumx-components-generic-block-genericblock--vertical-overflow)) I'll remove it.
It is activated by default on the content section though

Comment (Mcourt) [reply packages/lumx-react/src/components/generic-block/GenericBlock.stories.tsx:61]: Ah I see. Anyway, it was nitpicking :) Looks good to me.

**Comment by gcornut** [comment]
Test OK (100% coverage on unit test)

**Pull request merged**